tls/cfsslscan_handshake.go

```diff
@@ -2,9 +2,58 @@ package tls
 
 import "errors"
 
+// IsECcipherSuite returns whether a given uint16 is code for an
+// EC-based TLS cipher suite. Comments indicate non-EC ranges.
+// encodings are defined by IANA:
+// http://www.iana.org/assignments/tls-parameters/tls-parameters.xml
+func IsECcipherSuite(cs uint16) bool {
```

Comment:
This logic does look bizzare, and I agree that a table is cleaner. I did it this way because, originally (and as a hack) we were simply masking the cipherID against 0xc000, since that works for all the Go-TLS-supported suites and I was trying to extend that logic to cover all the cases. But you're right that using a table is the best solution, and I've ported the information over to CipherSuite and nuked this function.